src/meta/src/controller/streaming_job.rs

```diff
@@ -1514,6 +1514,51 @@ impl CatalogController {
         Ok(fragment_actors)
     }
 
+    async fn mutate_fragment_by_fragment_id(
```

Comment:
It seems only changing the fragment metadata, but no barrier is carrying out the information to the actors. Does it work for background ddl only?

Reply:
See: https://github.com/risingwavelabs/risingwave/pull/20795/files#diff-66d82e7ca80b658c018a0a5f6407e235d890ba46ecc677cdf83a626f4af7dde7R147-R174